internal/provider/job_resource.go

```diff
@@ -0,0 +1,333 @@
+	return ansibleVarsValidator{}
+}
+
+// Schema defines the schema for the resource.
+func (d *JobResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"job_template_id": schema.Int64Attribute{
+				Required: true,
+			},
+			"inventory_id": schema.Int64Attribute{
+				Optional: true,
+			},
+			"job_type": schema.StringAttribute{
+				Computed: true,
+			},
+			"job_url": schema.StringAttribute{
+				Computed: true,
+			},
+			"status": schema.StringAttribute{
+				Computed: true,
+			},
+			"extra_vars": schema.StringAttribute{
+				Optional:   true,
+				CustomType: jsontypes.NormalizedType{},
+			},
+			"ignored_fields": schema.ListAttribute{
+				ElementType: types.StringType,
+				Computed:    true,
+				Description: "The list of properties set by the user but ignored on server side.",
+			},
+		},
+	}
+}
+
+// jobResourceModel maps the resource schema data.
+type jobResourceModel struct {
+	TemplateID    types.Int64          `tfsdk:"job_template_id"`
+	Type          types.String         `tfsdk:"job_type"`
+	URL           types.String         `tfsdk:"job_url"`
+	Status        types.String         `tfsdk:"status"`
+	InventoryID   types.Int64          `tfsdk:"inventory_id"`
+	ExtraVars     jsontypes.Normalized `tfsdk:"extra_vars"`
+	IgnoredFields types.List           `tfsdk:"ignored_fields"`
+}
+
+var keyMapping = map[string]string{
+	"inventory":             "inventory",
+	"execution_environment": "execution_environment_id",
+}
+
+func (d *jobResourceModel) GetTemplateID() string {
+	return d.TemplateID.String()
+}
+
+func (d *jobResourceModel) GetURL() string {
+	if !d.URL.IsNull() && !d.URL.IsUnknown() {
+		return d.URL.ValueString()
+	}
+	return ""
+}
+
+func (d *jobResourceModel) ParseHTTPResponse(body []byte) error {
+	/* Unmarshal the json string */
+	var result map[string]interface{}
+	err := json.Unmarshal(body, &result)
+	if err != nil {
+		return err
+	}
+
+	d.Type = types.StringValue(result["job_type"].(string))
+	d.URL = types.StringValue(result["url"].(string))
+	d.Status = types.StringValue(result["status"].(string))
+	d.IgnoredFields = types.ListNull(types.StringType)
+
+	if value, ok := result["ignored_fields"]; ok {
+		var keysList = []attr.Value{}
+		for k := range value.(map[string]interface{}) {
+			key := k
+			if v, ok := keyMapping[k]; ok {
+				key = v
+			}
+			keysList = append(keysList, types.StringValue(key))
+		}
+		if len(keysList) > 0 {
+			d.IgnoredFields, _ = types.ListValue(types.StringType, keysList)
+		}
+	}
+
+	return nil
+}
+
+func IsValueProvided(value attr.Value) bool {
+	return !value.IsNull() && !value.IsUnknown()
+}
+
+func (d *jobResourceModel) CreateRequestBody() (io.Reader, diag.Diagnostics) {
+	body := make(map[string]interface{})
+
+	// Extra vars
+	var diags diag.Diagnostics
+	if IsValueProvided(d.ExtraVars) {
+		var extraVars map[string]interface{}
```

Comment:
Is it necessary to unmarshal this here?

Reply:
yes it is. TF stores only a raw value which is not JSON-parsed, this can generate errors

Reply:
But doesn't the normalized JSON type validator ensure that it's a valid JSON string?

Reply:
The normalized JSON type ensures that it is a valid JSON string. 
We can remove the unmarshall here, but this means the user should not escape quote into the JSON definition, the following code will be replaced by 

```golang
body["extra_vars"] = d.ExtraVars.ValueString()
```

and the main.tf will look like
```terraform
resource "aap_job" "sample" {
 ... 
 extra_vars = file("./vars.json")
  ...
}
```

with `vars.json`

```json
{ "some": "variable"}
```

In this case if the user configuration is like the following below, it will not work as expected

```terraform
resource "aap_job" "sample" {
 ... 
 extra_vars = "{ \"some\": \"variable\"}"
  ...
}
```

adding unmarshall here will resolve the issue